trieste/acquisition/function/greedy.py

```diff
@@ -351,3 +355,127 @@ def __call__(self, x: TensorType) -> TensorType:
         p = -5  # following experiments of :cite:`Alvi:2019`.
         penalization = ((pairwise_distances / (self._radius + self._scale)) ** p + 1) ** (1 / p)
         return tf.reduce_prod(penalization, axis=-1)
+
+
+class FantasizeAcquisitionFunction(GreedyAcquisitionFunctionBuilder):
+    r"""
+    Builder of the acquisition function maker for greedily collecting batches, following the
+    kriging believer heuristic.
```

Comment:
it would be easy to add constant liar as well, no?

Reply:
yes, although this heuristic doesn't come with a good default for the "liar" value. So it's probably not worth it for now?

The very flexible alternative would be to allow passing a callable. But probably an overkill?

Reply:
it does show good performance in comparison to KB, I think its worth including it
providing min, mean, max is straightforward, callable would be nice but more work